shell/platform/embedder/embedder.h

```diff
@@ -348,23 +348,33 @@ typedef enum {
   kFlutterSoftwarePixelFormatGray8,
 
   /// pixel with 5 bits red, 6 bits green, 5 bits blue, in 16-bit word.
-  ///   r = p & 0x3F; g = (p>>5) & 0x3F; b = p>>11;
+  ///   r = p>>11; g = (p>>5) & 0x3F; b = p & 0x3F;
+  ///
+  /// This is equivalent to wayland format RGB565 (WL_DRM_FORMAT_RGB565).
```

Comment:
If this is native endian, then it will only match DRM formats on little endian systems because they are always little endian. according to [this resource you linked on discord](https://afrantzis.com/pixel-format-guide/wayland_drm.html).

Also, given that this is for the software rendering backend, maybe it makes more sense to reference [the wl_shm formats](https://wayland.app/protocols/wayland#wl_shm:enum:format)? since `wayland-drm` is for EGL. Both are identical, I think? wl_shm formats are explicitly also based on Linux DRM codes and explicitly little endian. Maybe reference DRM codes instead of any Wayland protocol?

```suggestion
  /// On little endian systems, this is equivalent to wayland format RGB565 (WL_SHM_FORMAT_RGB565).
```

Reply:
> If this is native endian, then it will only match DRM formats on little endian systems because they are always little endian. according to [this resource you linked on discord](https://afrantzis.com/pixel-format-guide/wayland_drm.html).

True; though I'd also provide some context and say that big endian is super rare and in the vast majority of cases this is equivalent, for the lucky people that didn't have to think about endianess until now.

> Also, given that this is for the software rendering backend, maybe it makes more sense to reference [the wl_shm formats](https://wayland.app/protocols/wayland#wl_shm:enum:format)? since wayland-drm is for EGL. Both are identical, I think? wl_shm formats are explicitly also based on Linux DRM codes and explicitly little endian. Maybe reference DRM codes instead of any Wayland protocol?

Yes AFAICT they're both the based on the DRM codes. Though, I think still better to leave no room for confusion and give the precise wayland formats (maybe both WL_DRM_... and WL_SHM_...). I think if you're dealing with DRM directly you might know that WL_DRM_FORMAT_... also equals DRM_FORMAT_...